warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IsNotNullPruningVisitor.java

```diff
@@ -122,34 +117,36 @@ public static JexlNode prune(JexlNode node) {
      */
     @Override
     public Object visit(ASTAndNode node, Object data) {
+        // do not process marker nodes
```

Comment:
not traversing marker nodes may not be what we want here.  We were not doing that previously

Reply:
That's not exactly true. QueryPropertyMarkers are excluded via the visit to an ASTReference node in line 355
```
    @Override
    public Object visit(ASTReference node, Object data) {
        // do not descend into marker nodes
        if (!QueryPropertyMarkerVisitor.getInstance(node).isAnyType()) {
            node.childrenAccept(this, data);
        }
        return data;
    }
```

That said, this visitor should probably visit certain types of marker nodes when determining if an 'isNotNull' term can be pruned, for example:
```
# should prune. Value marker will put values for FOO into the context
!(FOO != null) && ((_Value_ = true) && (FOO =~ 'ba.*'))

# should prune. Delayed marker won't pull values for FOO into the context, but will exclude an event just as well as the isNotNull term.
!(FOO != null) && ((_Delayed_ = true) && (FOO =~ 'ba.*'))

# should NOT prune
!(FOO != null) && ((_Drop_ = true) && (FOO =~ 'ba.*'))
```

Reply:
https://github.com/NationalSecurityAgency/datawave/blob/integration/warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IsNotNullPruningVisitor.java#L357

I didn't make this logic up.  And nodes are no longer parented by reference nodes, so the reference node logic was added to the and node.  I am just trying to preserve existing behavior.  If we want to do something different we should write a new ticket.